src/util/macros.h

```diff
@@ -30,7 +30,7 @@ inline char* C_STRING(v8::Local<v8::String> string) {
 
 // Given a double, returns whether the number is a valid 32-bit signed integer.
 inline bool IS_32BIT_INT(double num) {
-	return floor(num) == num && num < 2147483648 && num >= -2147483648;
+	return floor(num) == num && num < 2147483648L && num >= -2147483647L - 1;
```

Comment:
Yep that works too, and I checked that it defaults to double unless you add the `f` (was sort of concerned it'd pick float by default.) That's probably cleaner than the `- 1`. Will update this.